src/blockCG.jl

```diff
@@ -106,10 +106,12 @@ for iter=1:maxIter
 	BLAS.gemm!('T','N',1.0,Q,Z,0.0,QTZ)
     Beta  = -pinvPTQ*QTZ
     
-    if ortho     
+    if ortho
         P,     =  mgs!(Z + P*Beta)
     else
-	    P     = Z  + P*Beta;
+		BLAS.gemm!('N','N',1.0,P,Beta,1.0,Z);
```

Comment:
OK - now it works. Improved the testing.

On Tue, Jun 7, 2016 at 11:21 AM, Eran Treister <erantreister@gmail.com>
wrote:

> Not so fast - the change doesn't work. With ortho==false, P and Z are the
> same vector... the test did not include this case.
>
> On Tue, Jun 7, 2016 at 11:13 AM, Lars Ruthotto <notifications@github.com>
> wrote:
>
>> In src/blockCG.jl
>> <https://github.com/lruthotto/KrylovMethods.jl/pull/10#discussion_r66123810>
>> :
>>
>> >          P,     =  mgs!(Z + P*Beta)
>> >      else
>> > -	    P     = Z  + P*Beta;
>> > +		BLAS.gemm!('N','N',1.0,P,Beta,1.0,Z);
>>
>> minor tweak. Can we pull line 112 before the if statement (line 108)? and
>> then do mgs!(Z) in line 110?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/lruthotto/KrylovMethods.jl/pull/10/files/5fca412d06b6b13e45641cf84113379b10761c6c#r66123810>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AO5HPUHEUJNxtGoX4Q_0Fj1S9vAXjobOks5qJbTUgaJpZM4IwM5i>
>> .
>>
>
>